osu.Game/Rulesets/Mods/ModChallenge.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Objects;
+using System.Collections.Generic;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Scoring;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Scoring;
+using System.ComponentModel;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public abstract class ModChallenge : ModFailCondition, IApplicableToBeatmap
+    {
+        public override ModType Type => ModType.DifficultyIncrease;
+        public override double ScoreMultiplier => 1;
+        public override bool RequiresConfiguration => true;
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(ModPerfect)).Append(typeof(ModEasyWithExtraLives)).ToArray();
```

Comment:
`ModPerfect` is listed as incompatible with `ModChallenge` in this file, but there's no corresponding change in the other direction in `ModPerfect` itself. (Which is extra surprising because `ModEasyWithExtraLives` - the other incompatible mod - *is* changed in this pull.)